feat: Preload editor settings within My Site

dcalhoun opened the pull request:
Improve editor load speed through preloading the editor settings. 

## Testing instructions

> [!Important]
> Enabling the `Experimental block editor` feature is required. 

**1: My Site without cache fetches editor settings**
1. Observe the network activity of the app.
1. Install and launch the app.
1. Switch sites. 
1. Verify a successful editor settings endpoint request is performed.

**2: My Site with cache does not fetch editor settings**
1. Switch to a previous site selected in test case 1. 
1. Verify an editor settings endpoint request is not performed. 

**3: Editor utilizes stale-while-revalidate caching**
1. Open the editor. 
2. Verify a successful editor settings endpoint request is performed.
3. Repeat steps one and two.

dcalhoun committed:
feat: Preload editor settings within My Site

Improve load speed of editor.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dcalhoun committed:
perf: Conditionally preload editor settings based on cache presence

Avoid redundant network requests triggered each time My Site is
displayed. The stale-while-revalidate approach remains used when the
editor is opened, meaning the latest editor settings are always fetched
on each editor open event.

dcalhoun committed:
task: Site selection utilizes latest GutenbergKit feature flag

Ensure the remote feature flag is taken into consideration for editor
settings fetching.

nbradbury reviewed (changes requested):
@dcalhoun This tested fine and code looks good, but there are failing tests that should be addressed.

dcalhoun committed:
test: Add necessary mocks for existing tests

References to feature flags and experimental features introduced test
failures.

dcalhoun committed:
test: Assert editor settings fetches

Assert new conditional logic based on experimental features.

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dcalhoun commented:
@nbradbury tests passing, branch rebased, and ready for review again. Thanks! 🙇🏻‍♂️

nbradbury reviewed (approved):
Looks good, feel free to merge when ready! :shipit:
